### PR TITLE
Revert "Allow precreation of AttributeSets for metrics (#1421)"

### LINKED ISSUE #1508: AttributeSet in the OpenTelemetry API
https://github.com/open-telemetry/opentelemetry-rust/pull/1421 allowed pre-creation of `AttributeSet` for scenarios requiring high performance. This also moved `AttributeSet` to the opentelemetry api. This has an _undesired/unintended_ side effect: The entire cost of creating `AttributeSet`is now paid in the API itself (like sort/hash/de-duplication), making it impossible to do optimizations in the SDK, as the perf hit is already paid in the API itself. 

This violates the core OTel requirement that the overhead of using the OTel API should be zero/close-to-zero, without an SDK configured. Because right now, even without an SDK configured, the OTel API does the `AttributeSet` creation which is a far from the zero/close-to-zero overhead. i.e the [NoOp ](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/noop.md) implementation is broken.

This is similar to https://github.com/open-telemetry/opentelemetry-rust/issues/1189, but this is much more serious due to the perf hit.

Need to find a way to achieve the intended perf gains, but yet move off entire `AttributeSet` creation to the actual SDK. Will discuss some ideas in the next community meeting.

cc : @lalitb @KallDrexx

### PR DESCRIPTION
**Pull request by KallDrexx**
This reverts commit 16fd1abb879c48fdb9c87e65e9aaabbe1afdf3bc.

Fixes #1508
Design discussion issue (if applicable) #

## Changes

Reverting the precached attribute sets to not incur costs at the API level.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Review by lalitb — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]
Thanks. 

Really sorry to have missed to spot the undesired side-effects on this earlier :( 
Hopefully we can reuse a lot of this work later.

**Comment by cijothomas** [comment]
Hi @open-telemetry/rust-approvers
I am merging this quickly without waiting for more approvals, as delaying would cause merge conflicts (there are upcoming changes touching same set of files). 
If they are concerns, please raise, and we can follow up.

**Pull request merged**
